### PR TITLE
Misc fixes - Temp fix PointerRelease not raised on iOS if subscribed to Tapped

### PR DESCRIPTION
**Pull request by dr1rrb**
* Temp fix `PointerRelease` event not raised on iOS if subscribed to `Tapped` (Mitigates: https://github.com/nventive/Uno/issues/864)
* Avoid useless warning when using enum `FlowDirection`
* Add dummy implementation of `CoreWindow.Get[Async]KeyState`

GitHub Issue (If applicable): #864

## PR Type
What kind of change does this PR introduce?
- Feature
- Other: mitigation

## What is the current behavior?
The `PointerReleased` is never fired on a `UIElement` if als subscribed to the `Tapped` event. The most visible effect of this, is that controls stay in pressed state.

## What is the new behavior?
The event is fired ... if we press long enough to trigger `TouchesBegan` (i.e. the `PointerPressed`) (cf. #864)

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information
This is only a mitigation : In some cases only the `Tapped` event is being raised without any pointer events. Is seems that the `GestureRecognizer` triggers fast enough to even prevent the `TouchesBegan` (and so the `Touches<Cancelled|Ended>`)

### CODE REVIEW

File: src/SamplesApp/UITests.Shared/UITests.Shared.projitems [hunk 1]
```diff
@@ -2633,381 +2642,7 @@
...
-    <Compile Update="C:\s\nv.github\uno\src\SamplesApp\UITests.Shared\Windows_UI_Xaml_Controls\CommandBar\CommandBar_Padding.xaml.cs">
-      <DependentUpon>CommandBar_Padding.xaml</DependentUpon>
-    </Compile>
-    <Compile Update="C:\s\nv.github\uno\src\SamplesApp\UITests.Shared\Windows_UI_Xaml_Controls\CommandBar\CommandBar_Uppercase_Resource.xaml.cs">
-      <DependentUpon>CommandBar_Uppercase_Resource.xaml</DependentUpon>
-    </Compile>
-    <Compile Update="C:\s\nv.github\uno\src\SamplesApp\UITests.Shared\Windows_UI_Xaml_Controls\CommandBar\CommandBar_With_Long_Sentences.xaml.cs">
-      <DependentUpon>CommandBar_With_Long_Sentences.xaml</DependentUpon>
-    </Compile>
-  </ItemGroup>
```
Comment (carldebilly) [inline src/SamplesApp/UITests.Shared/UITests.Shared.projitems]: Removed a lot of file nesting declarations...

File: src/Uno.UWP/UI/Core/CoreWindow.cs [hunk 2]
```diff
@@ -6,9 +6,17 @@
...
 {
-	public partial class CoreWindow 
+	public partial class CoreWindow
 	{
-		public global::Windows.UI.Core.CoreDispatcher Dispatcher
+		public CoreDispatcher Dispatcher
 			=> CoreDispatcher.Main;
+
+		[Uno.NotImplemented]
+		public CoreVirtualKeyStates GetAsyncKeyState(System.VirtualKey virtualKey)
```
Comment (carldebilly) [inline src/Uno.UWP/UI/Core/CoreWindow.cs]: Seems unrelated to the pr

File: src/Uno.UI/UI/Xaml/UIElement.iOS.cs [hunk 3]
```diff
@@ -459,9 +459,26 @@ public override void TouchesCancelled(NSSet touches, UIEvent evt)
...
 					OriginalSource = this
 				};
-				var isHandledInManaged = RaiseEvent(PointerCanceledEvent, args);
 
-				if (!isHandledInManaged)
+				// As soon as a gesture is recognized by a 'GestureRecognizer' (cf. Tap and DoubleTap events)
+				// the touches are "cancelled" and this method is invoked. So until we remove the GestureRecognizers
+				// for those events, we handle the "Cancelled" similarly as the "Ended" and we raise the "Release".
+				var pointerEventIsHandledInManaged = false;
+				if (IsPointerCaptured || evt.IsTouchInView(this))
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/UIElement.iOS.cs]: The test for this one is to put a Button inside a ScrollViewer, press on the Button, and then start to scroll. The Button's Click shouldn't be raised.

**Comment by davidjohnoliver** [comment]
Lots of unrelated files changed..

File: src/SamplesApp/UITests.Shared/UITests.Shared.projitems (hunk 1 above)
Comment (dr1rrb) [reply src/SamplesApp/UITests.Shared/UITests.Shared.projitems]: Removed a lot of invalid absolute path ;)

File: src/Uno.UWP/UI/Core/CoreWindow.cs (hunk 2 above)
Comment (dr1rrb) [reply src/Uno.UWP/UI/Core/CoreWindow.cs]: Actually this PR contains multiple minor changes made to support the SyncFusion DataGrid, however in comment I focused on the most important changes instead of writing useless comment like :
## Current Behavior
`VirtualKeyStates` throw exceptin

## New behavior
`VirtualKeyStates` always return `None`

File: src/Uno.UI/Generated/3.0.0.0/Windows.UI.Xaml/FlowDirection.cs [hunk 4]
```diff
@@ -2,7 +2,7 @@
 #pragma warning disable 114 // new keyword hiding
 namespace Windows.UI.Xaml
 {
-	#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __MACOS__
+	#if false
```
Comment (MatFillion) [inline src/Uno.UI/Generated/3.0.0.0/Windows.UI.Xaml/FlowDirection.cs]: im confused about this change. Why?

File: src/Uno.UWP/UI/Core/CoreWindow.cs (hunk 2 above)
Comment (dr1rrb) [reply src/Uno.UWP/UI/Core/CoreWindow.cs]: Splitted !

File: src/Uno.UI/Generated/3.0.0.0/Windows.UI.Xaml/FlowDirection.cs (hunk 4 above)
Comment (dr1rrb) [reply src/Uno.UI/Generated/3.0.0.0/Windows.UI.Xaml/FlowDirection.cs]: Expand the code ;) : one for the class, and one for the attribute. Actually this is generated code, I just manually updated it to match the newly implemented APIs

File: src/Uno.UI/UI/Xaml/UIElement.iOS.cs (hunk 3 above)
Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/UIElement.iOS.cs]: Did you make this test?

Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/UIElement.iOS.cs]: As the `Click` is linked to the `Tapped`, which is raised by the `GestureRecognizer`, it won't be affected by this commit. However yes I've validated, it works properly.

BTW I noticed that we don't even receive the `PointerPressed` nor `PointerReleased` on a `Button` on Windows (FYI @carldebilly ). This should be included in final fix of #864

**Comment by davidjohnoliver** [comment]
@dr1rrb  This is causing Button.Click to be fired twice (once from Tapped, then once from PointerReleased). This breaks CheckBox/ToggleButton amongst other things - IsChecked gets set to !(!IsChecked), which isn't very helpful. (I noticed because the menu button in SamplesApp stopped working.)

**Comment by davidjohnoliver** [comment]
And I just tried the Button-inside-ScrollViewer test, and Click *is* getting called!

**Comment by dr1rrb** [comment]
> 
> 
> And I just tried the Button-inside-ScrollViewer test, and Click _is_ getting called!

Actually it seems to depend of how long you press of the `Button` before "scrolling" and how planets are aligned (sometimes we get all events - possibly even a double `Click` - and sometimes we get the right event sequence). It's probably due to both the `GestureRecognizer` which [delays](https://developer.apple.com/documentation/uikit/uigesturerecognizer/1624209-delaystouchesended) and [alter](https://stackoverflow.com/questions/5818692/how-to-avoid-touches-cancelled-event) touches events, and to the `UIScrollView` which [adds a delay to propagate touches](https://developer.apple.com/documentation/uikit/uiscrollview/1619398-delayscontenttouches). So will revert this commit as it was only a workaround to avoid to fix the full issue (i.e. removing the gesture recognizer on iOS). 

Note: we will however create a custom branch with this commit (or a new commit which uses the flag [cancelstouchesinview](https://developer.apple.com/documentation/uikit/uigesturerecognizer/1624218-cancelstouchesinview) and/or a a feature flag ?) until we fix it properly.
